src/Routing/Router.php

```
@@ -924,4 +928,29 @@ public static function setRouteCollection(RouteCollection $routeCollection): voi
     {
         static::$_collection = $routeCollection;
     }
+
+    /**
+     * Inject route defaults from `_path` key
+     *
+     * @param array $url Route array with `_path` key
+     * @return array
+     */
+    protected static function unwrapShortString(array $url)
+    {
+        foreach (['plugin', 'prefix', 'controller', 'action'] as $key) {
+            if (array_key_exists($key, $url)) {
```

Wouldn't isset() suffice? As it shouldn't be non-null for it to become an issue right?

---

For isset() and possible key existance we might need

    $url += RouteBuilder::parseShortString($url['_path']);

to be

    $url = RouteBuilder::parseShortString($url['_path']) + $url;

though.

---

Should then `Router::url(['_path' => 'Cms.Articles::index', 'plugin' => null]);` silently ignore `null` and use plugin 'Cms'?

---

That depends on if null is the same meaning as false. The latter seems like a deliberate deactivation, whereas the null can be just default case from nullable setup, without meaning.
We might have to clarify what we want here as behavior of those two values.

---

I will then remove override check here, so it could be overridden. If there is no override, it shouldn't use current request.

`['_path' => 'Articles::index', 'action' => 'view]` same as `'Articles::view'`
`['_path' => 'Cms.Articles::index', 'plugin' => null]` same as `'Articles::index'`
`['_path' => 'Cms.Articles::index', 'prefix' => 'admin']` same as `'Cms.Admin/Articles::index'`

---

@markstory What do u think?

---

dont allow to override please

---

I left exception to not allow overriding. Only changed to `InvalidArgumentException`.

I replaced to `isset()` check. `null` overrides will be silently ignored.